airgun/browser.py

```diff
@@ -513,11 +513,16 @@ class AirgunBrowserPlugin(DefaultPlugin):
          spinner = document.getElementById("turbolinks-progress")
          return (spinner === null) ? true : spinner.style["display"] == "none"
         }
+        function reactLoadingInvisible() {
+         react = document.querySelector("#reactRoot .loading-state")
```

Comment:
Is whitespace really present before dot?

Reply:
Yes. I'm looking for element with class that is child of `#reactRoot`.

This is HTML when data is being loaded:

```html
<div id="reactRoot">
  <div>
    <div class="container-fluid">
      <div class="row">
        <div class="col-sm-12">
          <h1>Red Hat Subscriptions</h1>
          <div class="toolbar-pf table-view-pf-toolbar-external row">
            <div class="col-sm-12">
              <form class="toolbar-pf-actions">
                <div class="toolbar-pf-filter form-group">
                  <div><label class="control-label sr-only">Search</label>
                    <div><span class="input-group"><input role="combobox" aria-autocomplete="list" aria-expanded="false" autocomplete="off" type="text" id="downshift-0-input" class="form-control" value=""><span class="input-group-btn"><button type="button" class="btn btn-default">Search</button></span></span>
                    </div>
                  </div>
                </div>
                <div class="toolbar-pf-action-right">
                  <div class="form-group">
                    <div class="tooltip-button-helper"><a href="subscriptions/add" class="btn btn-primary disabled" tabindex="-1" style="pointer-events: none;">Add Subscriptions</a></div><button type="button" class="btn btn-default">Manage Manifest</button><button type="button" class="btn btn-default">Export CSV</button>
                    <div
                      class="tooltip-button-helper"><button disabled="" type="button" class="btn btn-danger">Delete</button></div>
                </div>
            </div>
            </form>
          </div>
        </div>
        <div id="subscriptions-table" class="modal-container">
          <div class="loading-state">
            <div class="spinner spinner-lg"></div>
            <p>Loading</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
</div>

</div>



<script defer="defer">
  //<![CDATA[
  $(tfm.reactMounter.mount('katello', '#reactRoot', []));
  //]]>
</script>
```

And after loading is finished:

```html
<div id="subscriptions-table" class="modal-container">
  <div>
    <table class="table table-striped table-bordered table-hover pf-table-inline-edit table-fixed">
      <thead>
        <tr class="">
          <th class="table-view-pf-select" aria-label="Select all rows"><label for="selectAll" class="control-label sr-only">Select all rows</label><input type="checkbox" id="selectAll"></th>
          <th class="">Name</th>
          <th class="">SKU</th>
          <th class="">Contract</th>
          <th class="">Start Date</th>
          <th class="">End Date</th>
          <th class="">Requires Virt-Who</th>
          <th class="">Consumed</th>
          <th class="">Entitlements</th>
        </tr>
      </thead>
      <tbody>
        <tr class="">
          <td class="table-view-pf-select"><label for="select0" class="control-label sr-only">Select row</label><input type="checkbox" id="select0"></td>
          <td><a href="/subscriptions/196/">Employee SKU</a></td>
          <td class="">ES0113909</td>
          <td class="">11341622</td>
          <td class="">2017-03-15 05:00:00 +0100</td>
          <td class="">2022-01-01 05:59:59 +0100</td>
          <td><span aria-hidden="true" class="fa fa-check"></span></td>
          <td class="">0</td>
          <td class="editable">
            <div class="input" role="textbox" tabindex="0">15</div>
          </td>
        </tr>
      </tbody>
    </table>
    <form class="content-view-pf-pagination table-view-pf-pagination clearfix">
      <div class="form-group">
        <div class="dropup pagination-pf-pagesize btn-group"><button id="pagination-row-dropdown" role="button" aria-haspopup="true" aria-expanded="false" type="button" class="dropdown-toggle btn btn-default">20 <span class="caret"></span></button>
          <ul role="menu" class="dropdown-menu" aria-labelledby="pagination-row-dropdown">
            <li role="presentation" class=""><a role="menuitem" tabindex="-1" href="#">5</a></li>
            <li role="presentation" class=""><a role="menuitem" tabindex="-1" href="#">10</a></li>
            <li role="presentation" class=""><a role="menuitem" tabindex="-1" href="#">15</a></li>
            <li role="presentation" class=""><a role="menuitem" tabindex="-1" href="#">25</a></li>
            <li role="presentation" class=""><a role="menuitem" tabindex="-1" href="#">50</a></li>
          </ul>
        </div><span>per page</span></div>
      <div class="form-group"><span><span class="pagination-pf-items-current">1-1</span>&nbsp;of&nbsp;<span class="pagination-pf-items-total">1</span></span>
        <ul class="pagination pagination-pf-back">
          <li class="disabled"><a href="#" title="First Page"><span aria-hidden="true" class="fa fa-angle-double-left i"></span></a></li>
          <li class="disabled"><a href="#" title="Previous Page"><span aria-hidden="true" class="fa fa-angle-left i"></span></a></li>
        </ul><label class="sr-only control-label">Current Page</label><input type="text" class="pagination-pf-page form-control" value="1"><span>&nbsp;of&nbsp;<span class="pagination-pf-pages">1</span></span>
        <ul class="pagination pagination-pf-forward">
          <li class="disabled"><a href="#" title="Next Page"><span aria-hidden="true" class="fa fa-angle-right i"></span></a></li>
          <li class="disabled"><a href="#" title="Last Page"><span aria-hidden="true" class="fa fa-angle-double-right i"></span></a></li>
        </ul>
      </div>
    </form>
  </div>
</div>
```

Reply:
@mirzal @oshtaier RHAI pages also have the same or similar spinner, at least part `<div class="spinner spinner-lg"></div>` is the same #187